src/input.rs

```diff
@@ -135,17 +135,12 @@ impl<'a> Input<'a> {
     }
 
     pub fn is_stdin(&self) -> bool {
-        if let InputKind::StdIn = self.kind {
-            true
-        } else {
-            false
-        }
+        matches!(self.kind, InputKind::StdIn)
```

Comment:
Similar here.
```
if InputKind::StdIn = self.kind {
```
should work just fine?

Reply:
Please set disable linters that you dont like. Like this now is confusing and annoying

Reply:
What do you mean? Why would we have to do that? It's not like we are enforcing zero clippy warnings anywhere.

Also, It's not that I don't like `clippy`. I think it's a great tool. But a linter does not *always* have the best suggestions. Which is why I added review comments to a few of your changes.

Reply:
@sharkdp me personally prefer the idiomatic rust way.
It does the same thing in (mostly) fewer lines of code, and there is no down|upgrade in perfomance; also warning are really annoying.

I love using this tool; and looking forward seeing new updates.

Reply:
> me personally prefer the idiomatic rust way

I, too, like idiomatic Rust. That's why I prefer
``` rust
     pub fn is_stdin(&self) -> bool { 
         self.kind == InputKind::StdIn
     } 
```
over
``` rust
     pub fn is_stdin(&self) -> bool { 
         matches!(self.kind, InputKind::StdIn)
     } 
```
Not only is it shorter, but also easier to read, especially for someone who doesn't know what `matches!(…)` does - which is rather new.

That's also why I prefer to have one single function that creates a `PrettyPrinter`. Either `default()`, or `new()`. Having both of them with the same functionality is not just confusing but also more code.

I added comments to a few of your changes. If you don't agree, please argue why you like your version better. Simply adding :-1: to all of my posts without giving any reasons is not really the greatest way to interact.

> also warning are really annoying

If there are compiler warnings, we should fix them, as they typically point to underlying problems. But what you are doing here is to simply copy & paste the `clippy` suggestion (*"help: try this: `matches!(self.kind, InputKind::StdIn)`"*), which is not always the best solution.